plaso/parsers/locate.py

```diff
@@ -0,0 +1,202 @@
+          modification time.
+
+    Raises:
+      UnableToParseFile: if the file cannot be parsed.
+    """
+    directory_header_map = self._GetDataTypeMap('directory_header')
+    directory_entry_map = self._GetDataTypeMap('directory_entry')
+    cstring_map = self._GetDataTypeMap('cstring')
+
+    while True:
+      if self._file_object.tell() + 16 > self._file_object.get_size():
+        break
+      try:
+        directory_header, directory_header_size = (
+            self._ReadStructureFromFileObject(self._file_object,
+            self._file_offset, directory_header_map))
+        self._file_offset += directory_header_size
+      except (ValueError, errors.ParseError) as exception:
+        raise errors.UnableToParseFile(
+          'Unable to parse locate directory header with error {0!s}'.format(
+            exception))
+
+      directory_timestamp = (
+          directory_header.time_sec*definitions.NANOSECONDS_PER_SECOND)
+      directory_timestamp += directory_header.time_nsec
+      posix_timestamp = posix_time.PosixTimeInNanoseconds(
+          timestamp=directory_timestamp)
+      yield directory_header.name, posix_timestamp
+
+      # skip over file / subdirectory names as they don't have any timestamps
+      while True:
+        try:
+          directory_entry, directory_entry_size = (
+              self._ReadStructureFromFileObject(
+                  self._file_object, self._file_offset, directory_entry_map))
+          self._file_offset += directory_entry_size
+          if directory_entry.type == 0x02:
+            break
+
+          _, data_size = self._ReadStructureFromFileObject(
+              self._file_object, self._file_offset, cstring_map)
+          self._file_offset += data_size
+        except (ValueError, errors.ParseError) as exception:
+          raise errors.UnableToParseFile(
+            'Unable to parse locate directory entry with error {0!s}'.format(
+              exception))
+
+
+  def Open(self, file_object):
+    """Opens a Locate database file.
+
+    Args:
+      file_object (dfvfs.FileIO): file-like object.
+
+    Raises:
+      IOError: if the file-like object cannot be read.
+      OSError: if the file-like object cannot be read.
+      ValueError: if the file-like object is missing.
+      UnableToParseFile: if the file cannot be parsed
+    """
+    if not file_object:
+      raise ValueError('Missing file object.')
+
+    self._file_object = file_object
+
+    locate_database_header_map = self._GetDataTypeMap('locate_database_header')
+
+    try:
+      locate_database_header, data_size = self._ReadStructureFromFileObject(
+          self._file_object, self._file_offset, locate_database_header_map)
+      self._file_offset += data_size
+    except (ValueError, errors.ParseError) as exception:
+      raise errors.UnableToParseFile(
+          'Unable to parse locate database header with error: {0!s}'.format(
+          exception))
+    if locate_database_header.signature != self.DB_MAGIC:
```

Comment:
informational: there is no need for this the signature is already checked since it is defined in GetFormatSpecification and in locate.yaml